### DISCUSSION TITLE
Spring boot 3 support for auto-configure

### DISCUSSION THREAD

**Discussion by swaroopar** [open-telemetry/opentelemetry-java-instrumentation#9928] [Q&A] [answered]
Hello All, 

In this [documentation page](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/instrumentation/spring/spring-boot-autoconfigure), it mentions to use `opentelemetry-spring-boot-starter` if we wish to use all features. 
But the [spring-boot-auto-configure](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/starters/spring-boot-starter/README.md) mentions that only spring-boot 2 is supported. 

My issue is that i cannot get the logs (using logback) exported. 
I used only 'opentelemetry-spring-boot-starter' dependency and the and the solution mentioned in https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9819, but still no logs were exported. Only traces and metrics were exported. 

```xml
<dependency>
            <groupId>io.opentelemetry.instrumentation</groupId>
            <artifactId>opentelemetry-spring-boot-starter</artifactId>
            <version>1.32.0-alpha</version>
        </dependency>

and enviroment variable OTEL_LOGS_EXPORTER=otlp
```

Could you please let me know if I must explicitly add all dependencies if I wish to use the collector with spring-boot version 3?

Thank you for all the great work.

**Comment by laurit** [comment]
Environment variables like `OTEL_LOGS_EXPORTER` only work when opentelemetry is set up via autoconfigure extension https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk-extensions/autoconfigure/README.md Spring boot starter does not do that, but that shouldn't be the issue because it enables the logging by default. You probably need to add `OpenTelemetryAppender` to your logging configuration, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/spring/spring-boot-autoconfigure/src/testLogbackAppender/resources/logback-test.xml

**Comment by swaroopar** [comment]
@laurit thank you for looking into my issue. I tried your suggestion but the logs are still not reaching the collector. 

I made the following changes to maven dependencies and logback config - https://github.com/swaroopar/osc/commit/dc0f223f25bd480db4c39c549724faca5ecb8c58

If I shutdown the collector docker container, I only see warnings for traces and metrics. Nothing for logs.
```
10:36:55.560 [OkHttp http://localhost:4317/...] ERROR i.o.e.i.grpc.OkHttpGrpcExporter []- Failed to export spans. The request could not be executed. Full error message: closed
10:37:30.518 [OkHttp http://localhost:4317/...] ERROR i.o.e.i.grpc.OkHttpGrpcExporter []- Failed to export metrics. The request could not be executed. Full error message: Failed to connect to localhost/127.0.1.1:4317
1
```

P.S - ~~could not use the latest version since I have to use same versions of all dependencies~~. Also the version mentioned in the documentation actually does not exist on maven central - https://opentelemetry.io/docs/instrumentation/java/automatic/spring-boot/#configuration
Updated all required dependencies to the latest version in POM. 

Please let me know if i'm still missing something.

**Comment by swaroopar** [comment]
@laurit after some debugging I saw that only the `NoopLogRecordProcessor` was getting activated. So added the below dependency and also retained all other otel dependencies with version1.32.0. Now the logger seem to be activated but fails with the below error. 

```xml
<dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-exporter-otlp-logs</artifactId>
            <version>1.26.0-alpha</version>
        </dependency>
```

```sh
Exception in thread "BatchLogRecordProcessor_WorkerThread-1" java.lang.NoSuchMethodError: 'long io.opentelemetry.sdk.logs.data.LogRecordData.getEpochNanos()'
	at io.opentelemetry.exporter.internal.otlp.logs.LogMarshaler.create(LogMarshaler.java:49)
	at io.opentelemetry.exporter.internal.marshal.MarshalerUtil.groupByResourceAndScope(MarshalerUtil.java:67)
	at io.opentelemetry.exporter.internal.otlp.logs.ResourceLogsMarshaler.groupByResourceAndScope(ResourceLogsMarshaler.java:98)
	at io.opentelemetry.exporter.internal.otlp.logs.ResourceLogsMarshaler.create(ResourceLogsMarshaler.java:38)
	at io.opentelemetry.exporter.internal.otlp.logs.LogsRequestMarshaler.create(LogsRequestMarshaler.java:33)
	at io.opentelemetry.exporter.otlp.logs.OtlpGrpcLogRecordExporter.export(OtlpGrpcLogRecordExporter.java:55)
	at io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor$Worker.exportCurrentBatch(BatchLogRecordProcessor.java:312)
	at io.opentelemetry.sdk.logs.export.BatchLogRecordProcessor$Worker.run(BatchLogRecordProcessor.java:229)
	at java.base/java.lang.Thread.run(Thread.java:1583)
```

**Comment by swaroopar** [comment, marked as answer]
Adding this specific dependency is needed for spring-boot in addition to the starter 

```xml
<dependencyManagement>
    <dependencies>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-bom</artifactId>
            <version>1.32.0</version>
            <type>pom</type>
        </dependency>
    </dependencies>
</dependencyManagement>
```
